Packages/com.unity.inputsystem/InputSystem/Plugins/HID/HIDSupport.cs

```diff
@@ -28,29 +28,64 @@ namespace UnityEngine.InputSystem.HID
+        /// </remarks>
+        public struct HIDPageUsage
         {
-            add => s_ShouldCreateHID.Append(value);
-            remove => s_ShouldCreateHID.Remove(value);
-        }
+            /// <summary>
+            /// The usage page.
+            /// </summary>
+            public HID.UsagePage page;
 
-        internal static InlinedArray<ShouldCreateHIDCallback> s_ShouldCreateHID;
+            /// <summary>
+            /// A number specifying the usage on the usage page.
+            /// </summary>
+            public int usage;
 
-        private static bool? DefaultShouldCreateHIDCallback(HID.HIDDeviceDescriptor descriptor)
-        {
-            if (descriptor.usagePage == HID.UsagePage.GenericDesktop)
+            /// <summary>
+            /// Create a HIDPageUsage struct by specifying a page and usage.
+            /// </summary>
+            public HIDPageUsage(HID.UsagePage page, int usage)
             {
-                switch (descriptor.usage)
-                {
-                    case (int)HID.GenericDesktop.Joystick:
-                    case (int)HID.GenericDesktop.Gamepad:
-                    case (int)HID.GenericDesktop.MultiAxisController:
-                        return true;
-                }
+                this.page = page;
+                this.usage = usage;
+            }
+
+            /// <summary>
+            /// Create a HIDPageUsage struct from the GenericDesktop usage page by specifying the usage.
+            /// </summary>
+            public HIDPageUsage(HID.GenericDesktop usage)
+            {
+                this.page = HID.UsagePage.GenericDesktop;
+                this.usage = (int)usage;
             }
-            return null;
         }
 
+
+        /// <summary>
+        /// An array of HID usages the input is configured to support.
+        /// </summary>
+        /// <remarks>
+        /// The input system will only create <see cref="InputDevice"/>s for HIDs with usages
+        /// listed in this array. Any other HID will be ignored. This saves the input system from
+        /// spending resources on creating layouts and devices for HIDs which are not supported or
+        /// not usable for game input.
+        ///
+        /// By default, this includes only <see cref="HID.GenericDesktop.Joystick"/>,
+        /// <see cref="HID.GenericDesktop.Gamepad"/> and <see cref="HID.GenericDesktop.MultiAxisController"/>,
+        /// but you can set this property to include any other HID usages.
+        ///
+        /// Note that currently on macOS, the only HID usages which can be enabled are
+        /// <see cref="HID.GenericDesktop.Joystick"/>, <see cref="HID.GenericDesktop.Gamepad"/>,
+        /// <see cref="HID.GenericDesktop.MultiAxisController"/>, <see cref="HID.GenericDesktop.TabletPCControls"/>,
+        /// and <see cref="HID.GenericDesktop.AssistiveControl"/>.
+        /// </remarks>
+        public static ReadOnlyArray<HIDPageUsage> supportedHIDUsages { get; set; }
```

Comment:
I think the setter should make a private copy of the data. Otherwise, the array is basically readwrite for the caller.

Reply:
@Rene-Damm I'm not sure I understand? Maybe I don't get how ReadOnlyArray works, but how would the caller write to the array?

Reply:
```
var usages = new HIDPageUsage[10];
HIDSupport.supportedHIDUsages = usages; // implicit conversion to ReadOnlyArray
usages[0] = ...;
```

Overall, `ReadOnlyArray` pretends to be more than it really is. Was recently trying to find a .NET standard type that could replace the thing now that we're finally in .NET 4.5 land but unfortunately, nothing seems to quite do exactly what we want :/

In the case here, with the assigning of ReadOnlyArray, the writability IMO isn't that huge a deal. From my POV, more about the API behaving in a predictable way and not allowing a scenario where the data can suddenly change behind your back.

On the other hand, ReadOnlyArray itself can lead to some surprising effects. These days I'm thinking ReadOnlyArray should really be a ref struct. Anyway, digressing.

Reply:
Oh, I see. I didn't realize that when constructing a ReadOnlyArray from an array, it would keep the source array as backing storage. Yeah, makes complete sense now, I will change it.

Reply:
Fixed.

Reply:
Oops, backing array should have been private, fixing.